src/Generation/ResourcesGenerator.php

```diff
@@ -172,15 +172,21 @@ public static function generateDescriptorConfig(ServiceDetails $serviceDetails,
                 ->toArray(fn($x) => $x->getNameCamelCase(), fn($x) => $x->getPattern());
         }
 
-        $return = AST::return(
+        $codeBlock = AST::return(
             AST::array([
                 'interfaces' => AST::array([
                     $serviceDetails->serviceName => AST::array($serviceDescriptor)
                 ])
             ])
         );
 
-        return "<?php\n\n{$return->toCode()};";
+        $currentYear = (int)date("Y");
+
+        return AST::file(null)
+            ->withApacheLicense($currentYear)
+            ->withGeneratedCodeWarning()
+            ->withBlock($codeBlock)
+            ->toCode() . ";"; 
```

Comment:
I am adding the `;` manually here but I am not particularly fond of this. 

My first attempt was to add this to the `return` method on `AST` but that function is used in different places. I am using the `PhpFile` class which has the ability to use a codeblock but any `PhpBlock` class implements the `ShouldNotApplySemicolonInterface` which the `AST::toPhp` method checks to decided wether or not add a semicolon.

We could create a `EndPhpBlock` or `FinalPhpBlock` class that does not implements the interface perhaps?
```php
AST::file(null)
    ->withBlock(new EndPhpBlock($string))
```

 Add an optional parameter to force the semicolon?:
```php
AST::file(null)
    ->withBlock(new PhpBlock($string))
    ->withSemiColon()
```

perhaps a method on the `PhpClass` to "finalize" the class? But perhaps the word for "Final" might be confused by a Final class. Ready perhaps? that adds a `;` to the end?
```php
AST::file(null)
    ->withBlock(new PhpBlock())
    ->ready()
```

Reply:
Interesting...I don't want to mess with `PhpBlock` too much, especially if it is already opinionated about semicolons. Given how choosy the `AST` is about semicolons altogether, I think this might be a reasonable solution to keep it how it is...or do what you suggested with a new `EndPhpBlock`...maybe `LastPhpBlock`? It just wraps a `Block` and tacks on the `;` in `toCode`?

Reply:
I poked around a bit more today to see if adding the extra PHP class would help but after debugging and poking further turns out that in this case, it is actually not using the `phpBlock` class, but just a regular `AST` class and the `toPhp` method just calls `toCode` on an AST instance:
```php
elseif ($x instanceof AST) {
    return $x->toCode();
}
```

So the semicolon interface does nothing in this case lol. I think it would be best to have a way to add a semicolon codewise but for now let's keep it like it is.

Reply:
> So the semicolon interface does nothing in this case lol. I think it would be best to have a way to add a semicolon codewise but for now let's keep it like it is.

Nice code spelunking! I agree, simple enough for now. Maybe file a `priority: p3` `type: cleanup` bug to look at this?